src/NetEscapades.AspNetCore.SecurityHeaders/SecurityHeadersMiddleware.cs

```diff
@@ -96,7 +96,7 @@ private static Task OnResponseStarting(object state)
         IReadOnlyHeaderPolicyCollection? policyToApply = null;
         if (options.PolicySelector is not null)
         {
-            policyToApply = options.PolicySelector(
+            policyToApply = await options.PolicySelector(
```

Comment:
Would definitely want to refactor this to avoid needing to await if it's not required 🤔